### DISCUSSION TITLE
Exemplo de URL válida e real para testes de Pix dinâmicos

### DISCUSSION THREAD

**Discussion by pedrofurtado** [bacen/pix-api#424] [Ideas]
Olá pessoal!

Existe, na documentação oficial do Bacen ou em outro local, exemplos de URLs válidas e reais para testes de Pix dinâmicos?

Pix estáticos são mais fáceis de testar, justamente por sua natureza estática (única informação que precisa ser real nele é a chave Pix).

Contudo, no caso de Pix dinâmicos, me surgiu essa questão.

Suponho que só seja possível o teste mediante pagamento de uma integração com PSP, mas gostaria de confirmar de todo modo.

Obrigado!

**Comment by rafaelchagasb** [comment]
@pedrofurtado O Bacen possui uma ferramenta onde ele disponibiliza qrcodes de teste. Se a sua instituição for um participante direto acho que basta solicitar o acesso.

https://qr-h.sandbox.pix.bcb.gov.br/

Uma outra ferramenta não oficial que uso é a https://pix.nascent.com.br/tools/pix-qr-decoder/. Porém ela necessita que você tenha a url de location do qrcode e nesse caso precisaria de algum parceiro para gerar.

**Comment by pedrofurtado** [comment]
Obrigado pelo retorno @rafaelchagasb !

Infelizmente no meu caso não pertenço a uma instituição participante (PsP), eu estava em busca de URLs válidas para incorporar nos testes de uma biblioteca open-source que mantenho, que tem por objetivo justamente gerar QRCodes e BRCodes para Pix estáticos e dinâmicos, com linguagem Ruby. Os estáticos deu super certo, resta agora o teste "pra valer" com os dinâmicos, mas acabei não dando prosseguimento por conta dessa questão da URL.

A título de curiosidade, a biblioteca em questão é essa aqui: https://github.com/pedrofurtado/qrcode_pix_ruby/

Não há qualquer possibilidade de gerar URLs pra teste, imagino ne? Salvo exceção se formos um participante direto, certo? 🤝

**Comment by pedrofurtado** [comment]
Por enquanto, testamos com URLs "genéricas", mas fato é que com uma URL real (mesmo que de homologação, mas real no sentido de debitar e os app dos bancos identificarem e etc) seria jóia 🤩

**Comment by rafaelchagasb** [comment]
@pedrofurtado  Existe uma possibilidade. 

Alguns PSPs estão ofertando a APi Pix um ambiente Sandbox ou de homologação. Você pode procurar o site de Dev de algumas das grandes instituições para ver os passos para utilizar.

Pela Api Pix você pode gerar uma cobrança imediata(cob) ou cobrança com vencimento(cobv). Ambas irão retornar um atributo chamado location que é o que você coloca na tela do Create Pix do site https://pix.nascent.com.br/tools/pix-qr-decoder/.

Alguns PSPs já estão retornando o próprio qrcode em algum atributo específico ou até em outro endpoint.

**Comment by pedrofurtado** [comment]
Não sabia dessa possibilidade. Legal!

É possível você me informar algum PsP que tenha esse serviço sandbox de maneira gratuita? Ou não é permitido?

**Comment by fanpulse** [comment]
No processo de homologação do PIX para participante direto, a URL que irá aparecer no qr tester é da própria IF ?
Ou é uma URL de outro parceiro?
Resumo, para homologação eu preciso disponibilizar os qrcode gerados pelo participante do teste?
